sycl-compress/CMakeLists.txt

```diff
@@ -0,0 +1,86 @@
+cmake_minimum_required(VERSION 3.20.0)
+
+set(SYCL_COMPRESS_VERSION 0.0.7)
+project (sycl-compress VERSION "${SYCL_COMPRESS_VERSION}" LANGUAGES CXX)
+
+# Setting the same version as SYCL
+set(CMAKE_CXX_STANDARD 17)
+
+set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Build type (default Release)" FORCE)
+
+set(SYCL_COMPRESS_INCLUDE_DIR ${CMAKE_CURRENT_LIST_DIR})
+set(CMAKE_BINARY_DIR ${CMAKE_SOURCE_DIR}/lib/${CMAKE_BUILD_TYPE})
+set(LIBRARY_OUTPUT_PATH ${CMAKE_BINARY_DIR})
+
+# Download and build zstd statically.
+set(ZSTD_INSTALL_DIR ${CMAKE_CURRENT_BINARY_DIR}/zstd/install)
+
+if (WIN32)
+  set(ZSTD_LIBRARY_NAME zstd_static.lib)
+else()
+  set(ZSTD_LIBRARY_NAME libzstd.a)
+endif()
+
+include(ExternalProject)
+ExternalProject_Add(zstd
+  DEPENDS
+  URL https://github.com/facebook/zstd/releases/download/v1.5.6/zstd-1.5.6.tar.gz
```

Comment:
@uditagarwal97, please, check if the license of zstd code allows using it with our compiler.
@stdale-intel should be able to assist if needed.